### PR TITLE
Add github action to rebase pull request automatically

### PR DESCRIPTION
**Pull request by JaxLinAMD**

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 0c0291461c8a8f28716758f85b5f6a05c0620d0e
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 9786fa8c34df6d9baddeff40d331106799fcbb07</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev 9b77f6008d4316922cde45baf39d079b4d328893</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 1b94c3bf02ef6b638e199aa5c41d2fa31ea5455d</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225557)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30786/56392 (54.6%)</li>
<li>Failed: 0/56392 (0.0%)</li>
<li>Not Supported: 25606/56392 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30785/56392 (54.6%)</li>
<li>Failed: 0/56392 (0.0%)</li>
<li>Not Supported: 25607/56392 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30917/56381 (54.8%)</li>
<li>Failed: 0/56381 (0.0%)</li>
<li>Not Supported: 25464/56381 (45.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34387/56392 (61.0%)</li>
<li>Failed: 0/56392 (0.0%)</li>
<li>Not Supported: 22005/56392 (39.0%)</li>
</ul></details>


</ul></details>

File: .github/workflows/rebase.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Automatic Rebase
+on:
+  issue_comment:
+    types: [created]
+jobs:
+  rebase:
+    name: Rebase
+    if: github.event.issue.pull_request != '' && contains(github.event.comment.body, '/rebase')
```
Comment (kuhar) [inline .github/workflows/rebase.yml:8]:
```suggestion
    if: github.event.issue.pull_request != '' && contains(github.event.comment.body, '/rebase') &&
        (github.event.comment.author_association == 'OWNER' ||
         github.event.comment.author_association == 'COLLABORATOR')
```

**Review by kuhar — COMMENTED** [review]
I tested this on my fork and the new action seems to work well with the code review style used on this repo:
- When there's nothing to rebase, nothing happens
- PRs with multiple stacked commits are handled fine when unrelated commits land on `dev`
- PRs with multiple stack commits are handled fine when some of the child commits land on `dev`
- With the modification that I suggested, random people won't be able to spam us with rebases

**Review by kuhar — APPROVED** [review]

**Comment by kuhar** [comment]
@JaxLinAMD I tried to run it on Steven's PR and I think the issues is that it actually pushes updated commits to forks instead of the main repo (GPUpen-Drivers). I think it will start working one these forks have this workflow on their side (assuming they have Actions enabled), so that the workflow protection does not kick in.

E.g.: https://github.com/GPUOpen-Drivers/llpc/runs/2742764697?check_suite_focus=true
```
Successfully rebased and updated refs/heads/geometry_refactor.
+ git push --force-with-lease fork geometry_refactor
To https://github.com/s-perron/llpc.git
 ! [remote rejected]   geometry_refactor -> geometry_refactor (refusing to allow a GitHub App to create or update workflow `.github/workflows/rebase.yml` without `workflows` permission)
error: failed to push some refs to 'https://github.com/s-perron/llpc.git'
```

**Comment by amdvlk-admin** [comment]
### Test summary for commit cf1af650a8e637f2f4be2ae1528c365699542252
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 7387247eb9889ddcabbc1053b9c2052e253b088e</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev 9786fa8c34df6d9baddeff40d331106799fcbb07</li>
<li>SPVGEN: dev 5bf4618ac00d4675f57ca5b571a485bbc4d4fa33</li>
<li>XGL: dev 9b77f6008d4316922cde45baf39d079b4d328893</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 7da5ed485404334756d765594dd151e1ad31af55</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225557)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 30786/56392 (54.6%)</li>
<li>Failed: 0/56392 (0.0%)</li>
<li>Not Supported: 25606/56392 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 30785/56392 (54.6%)</li>
<li>Failed: 0/56392 (0.0%)</li>
<li>Not Supported: 25607/56392 (45.4%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 30917/56381 (54.8%)</li>
<li>Failed: 0/56381 (0.0%)</li>
<li>Not Supported: 25464/56381 (45.2%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 34387/56392 (61.0%)</li>
<li>Failed: 0/56392 (0.0%)</li>
<li>Not Supported: 22005/56392 (39.0%)</li>
</ul></details>


</ul></details>

**Comment by JaxLinAMD** [comment]
ok, thanks, I will keep an eye on it.

**Pull request merged**